server/src/test/java/org/elasticsearch/index/mapper/DynamicMappingTests.java

```diff
@@ -942,4 +947,48 @@ public void testSubobjectsFalseRootDynamicUpdate() throws Exception {
 
         assertEquals(0, mapperService.mappingLookup().objectMappers().size());
     }
+
+    public void testDefaultDenseVectorMappings() throws IOException {
+        DocumentMapper mapper = createDocumentMapper(topMapping(b -> b.field("numeric_detection", true)));
+        doTestDefaultDenseVectorMappings(mapper, XContentFactory.jsonBuilder());
+        doTestDefaultDenseVectorMappings(mapper, XContentFactory.yamlBuilder());
+        doTestDefaultDenseVectorMappings(mapper, XContentFactory.smileBuilder());
+        doTestDefaultDenseVectorMappings(mapper, XContentFactory.cborBuilder());
+    }
+
+    private void doTestDefaultDenseVectorMappings(DocumentMapper mapper, XContentBuilder builder) throws IOException {
+        BytesReference source = BytesReference.bytes(
+            builder.startObject()
+                .field("mapsToFloatTooSmall", Randomness.get().doubles(MIN_DIMS_FOR_DYNAMIC_FLOAT_MAPPING - 1, 0.0, 5.0).toArray())
+                .field("mapsToFloatTooBig", Randomness.get().doubles(MAX_DIMS_COUNT + 1, 0.0, 5.0).toArray())
+                .field("mapsToDenseVector", Randomness.get().doubles(MIN_DIMS_FOR_DYNAMIC_FLOAT_MAPPING, 0.0, 5.0).toArray())
+                .endObject()
+        );
+        ParsedDocument parsedDocument = mapper.parse(new SourceToParse("id", source, builder.contentType()));
+        Mapping update = parsedDocument.dynamicMappingsUpdate();
+        assertNotNull(update);
+        assertThat(((FieldMapper) update.getRoot().getMapper("mapsToFloatTooSmall")).fieldType().typeName(), equalTo("float"));
+        assertThat(((FieldMapper) update.getRoot().getMapper("mapsToFloatTooBig")).fieldType().typeName(), equalTo("float"));
+        assertThat(((FieldMapper) update.getRoot().getMapper("mapsToDenseVector")).fieldType().typeName(), equalTo("dense_vector"));
+        DenseVectorFieldMapper dvFieldMapper = ((DenseVectorFieldMapper) update.getRoot().getMapper("mapsToDenseVector"));
+    }
+
+    public void testDefaultDenseVectorMappingsObject() throws IOException {
+        DocumentMapper mapper = createDocumentMapper(topMapping(b -> b.field("numeric_detection", true)));
+        BytesReference source = BytesReference.bytes(
+            XContentFactory.jsonBuilder()
+                .startObject()
+                .startObject("parent_object")
+                .field("mapsToFloatTooSmall", Randomness.get().doubles(MIN_DIMS_FOR_DYNAMIC_FLOAT_MAPPING - 1, 0.0, 5.0).toArray())
+                .field("mapsToFloatTooBig", Randomness.get().doubles(MAX_DIMS_COUNT + 1, 0.0, 5.0).toArray())
+                .field("mapsToDenseVector", Randomness.get().doubles(MIN_DIMS_FOR_DYNAMIC_FLOAT_MAPPING, 0.0, 5.0).toArray())
```

Comment:
Nit, a little bit shorter:

```suggestion
                .field("mapsToFloatTooSmall", random().doubles(MIN_DIMS_FOR_DYNAMIC_FLOAT_MAPPING - 1, 0.0, 5.0).toArray())
                .field("mapsToFloatTooBig", random().doubles(MAX_DIMS_COUNT + 1, 0.0, 5.0).toArray())
                .field("mapsToDenseVector", random().doubles(MIN_DIMS_FOR_DYNAMIC_FLOAT_MAPPING, 0.0, 5.0).toArray())
```

Reply:
I started with that, but found that it's not allowed - it's on the forbidden methods list when you run `check`.

Reply:
Thanks for the info! 👍